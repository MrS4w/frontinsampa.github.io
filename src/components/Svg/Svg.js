import React from 'react';
import PropTypes from 'prop-types';

import InlineSvg from 'react-inlinesvg';

import Image from '../Image';

const Svg = ({ src, fallback }) => (
  <InlineSvg src={src}>
    <Image src={fallback} alt="" presentation />
  </InlineSvg>
);

Svg.propTypes = {
  src: PropTypes.string.isRequired,
  fallback: PropTypes.string.isRequired,
};

export default Svg;
