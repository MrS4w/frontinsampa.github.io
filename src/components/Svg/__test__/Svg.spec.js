import React from 'react';
import { shallow } from 'enzyme';

import Svg from '../Svg';

describe('Components: <Svg />', () => {
  const wrapper = shallow(
    <Svg src="/path/to/image.svg" fallback="/path/to/image.png" />,
  );

  it('Create a snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
