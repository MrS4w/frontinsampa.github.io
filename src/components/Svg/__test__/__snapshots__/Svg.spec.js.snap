// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <Svg /> Create a snapshot. 1`] = `
<InlineSVG
  cacheRequests={true}
  src="/path/to/image.svg"
  uniquifyIDs={false}
>
  <Image
    a11y={null}
    alt=""
    height="auto"
    presentation={true}
    src="/path/to/image.png"
    width="auto"
  />
</InlineSVG>
`;
